storage/http_client.go

```diff
@@ -302,7 +302,23 @@ func (c *httpStorageClient) UpdateBucket(ctx context.Context, bucket string, uat
 }
 
 func (c *httpStorageClient) LockBucketRetentionPolicy(ctx context.Context, bucket string, conds *BucketConditions, opts ...storageOption) error {
-	return errMethodNotSupported
+	s := callSettings(c.settings, opts...)
+
+	var metageneration int64
+	if conds != nil {
+		metageneration = conds.MetagenerationMatch
+	}
+	req := c.raw.Buckets.LockRetentionPolicy(bucket, metageneration)
+
+	setClientHeader(req.Header())
+	if s.userProject != "" {
+		req.UserProject(s.userProject)
+	}
```

Comment:
L313-316 is not in the original `BucketHandle.LockRetentionPolicy`; does it need to be included here? Since this is an admin operation, I'm not sure that Requester Pays is really relevant.

Reply:
It's your call, I just assumed (and assumed incorrectly) :) What say ye?

Reply:
Let's keep it consistent with what's in the http client currently. Can always make changes later if need be.

Reply:
SGTM! Removed!